storage_mgr.c

```diff
@@ -160,4 +160,39 @@ extern RC destroyPageFile(char *fileName)
     printError(*RC_message);
     return returnCode;
   }
-}
\ No newline at end of file
+}
+
+//Author: Rana Feyza Soylu
+extern RC readBlock (int pageNum, SM_FileHandle *fHandle, SM_PageHandle memPage)
+{
+  if (fHandle == NULL){ //Making sure file has been initialized/opened
+    return RC_FILE_HANDLE_NOT_INIT;
+  }
+  if (fHandle->totalNumPages - 1 < pageNum || pageNum < 0){
+    return RC_READ_NON_EXISTING_PAGE;
+  } else {
+    //Placing the pointer at the start of the block that we want to read
+    fseek(fHandle->mgmtInfo,pageNum*PAGE_SIZE,SEEK_SET); //SEEK_SET starts it from the beginning of the file, the offset is the total amount we need to move to get to pageNum
```

Comment:
Wrap this in an Error Block. Might throw Mem error if fseek fails.

Reply:
Resolved!